skfuzzy/__init__.py

```diff
@@ -71,14 +70,21 @@
 data_dir = osp.join(pkg_dir, 'data')
 
 try:
-    imp.find_module('nose')
+    # py3
+    import importlib.util
+    importlib.import_module('nose')
 except ImportError:
-    def _test(doctest=False, verbose=False):
-        """This would run all unit tests, but nose couldn't be
-        imported so the test suite can not run.
-        """
-        raise ImportError("Could not load nose. Unit tests not available.")
-
+    try:
+        # py2
+        import imp
+        imp.find_module('nose')
+        del imp
+    except ImportError:
+        def _test(doctest=False, verbose=False):
+            """This would run all unit tests, but nose couldn't be
+            imported so the test suite can not run.
+            """
+            raise ImportError("Could not load nose. Unit tests not available.")
```

Comment:
The remaining error is due to this new nesting not defining `_test()` before it is used in line 119 in Python 2; previously this was defined in the `else` of the `try-except-else` after the exception clause isn't triggered (Python 2 branch).

Under the new structure in Python 2, the top-level `else` does not execute and `_test` is not defined when used.  Simplified example:

``` python
try:
    print("first try statement")
    # force Py2 path
    import notamodule
except ImportError:
    try:
        print("python 2, nose available")
        # import notamodule  # Uncomment to simulate Py2 without Nose available (which works)
    except ImportError:
        print("_test defined")
else:
    print("_test defined")
finally:
    print("cleanup")
```

One solution - though perhaps not the most elegant - is to move the definition of `_test` into the `finally` clause and make it conditional on `_test` not existing or with a try/except `NameError`.  

Could also have both different `_test` functions defined at the top level and use `test = _test_withnose` or `test = _test_nonose` in these structures, which might be the most elegant.

Reply:
Ah! Good spot! Sorry. Updated PR using the second option :+1: Thanks!